libs/frontend/feature-command/src/lib/component/remove-component.ts

```diff
@@ -45,8 +45,17 @@ export const applyRemoveComponent = (site: ISite, data: IRemoveComponentData) =>
 }
 
 // We can almost reuse `applyAddComponent`, but we don't want to generate a new ID
-// and we need to recover the children
+// and we need to recover the children.
 export const undoRemoveComponent = (site: ISite, data: IRemoveComponentData) => {
+  undoRemoveComponentHelper(site, data, true)
+}
+
+// Declare a new function so that we don't have to add additional arguments to `undoRemoveComponent`.
+export const undoRemoveComponentHelper = (
+  site: ISite,
+  data: IRemoveComponentData,
+  selectAfterUndo: boolean,
+) => {
```

Comment:
I have to make a new function for undoing remove component because I need a new argument to determine if we should automatically select the restored component after undo.

The main reason is when undoing replace page root, old components (old root and its children) will be added back by using `undoRemoveComponent` command. The auto-select affected component feature in `undoRemoveComponent` will try to find which page does the restored component belong to, switch to that page & select that component. But at this moment, the root of the replaced page has not been restored yet (because the undo remove is executed bottom-up), so an error will be thrown. To avoid this problem I decided to make an almost identical function as the original `undoRemoveComponent`, but provide the third argument `selectAfterUndo` so that we don't have to add additional argument on `undoRemoveComponent`, which is used in `undoCommand` helper.

This sounds very complicated; I think I didn't explain it very well 😬. Feel free to let me know if you need further explanation!

Reply:
It makes sense! I've wondered if selecting/unselecting components within commands is good, since it's the only "side effect" (non-`site.context` change), but I'm not sure how else to do it.